daemon.go

```diff
@@ -49,6 +49,8 @@ const (
 	dhtSource  = "Amino DHT"
 )
 
+var defaultProtocolFilter = []string{"transport-bitswap", "unknown"}
```

Comment:
nit: we will be blind to `transport-ipfs-gateway-http` providers which are useful in browser already (eg. https://inbrowser.link/), and we want to support in GO as well.

Example: https://delegated-ipfs.dev/routing/v1/providers/bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi has `transport-ipfs-gateway-http` providers that do not speak bitswap.

I think we need to expose the filters via GUI, just like one can adjust IPNI URL + also probe for HTTP, as its useful for Helia.

We could ship as-is and resolve remaining work in separate PR, but at least add TODO:

```suggestion
// TODO: make this configurable, and add support and trustless retrieval probe for transport-ipfs-gateway-http
var defaultProtocolFilter = []string{"transport-bitswap", "unknown"}
```